### ISSUE TITLE
Rancher fails to build dialer to node in HA Airgap setup

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#16490]
Rancher/rancher:v2.1.2-rc7 
HA Airgap setup

When adding a custom cluster, the cluster failed to provision. 
1 node, all roles

* Deployed Rancher server in HA air gap
* Added custom cluster
* Registered node without specifying public/private ip in node registration command
* Cluster failed to provision saying could not connect to the node

Rancher logs:
```
2018/11/07 20:33:58 [INFO] cluster [c-qtv9m] provisioning: [dialer] Setup tunnel for host [172.31.24.13]
2018/11/07 20:35:38 [ERROR] cluster [c-7cz2m] provisioning: Failed to set up SSH tunneling for host [172.31.24.13]: Can't establish dialer connection: can not build dialer to c-7cz2m:m-1829ca4f89b2
```

Kubelet log:
```
+ docker start kubelet
Error response from daemon: {"message":"No such container: kubelet"}
Error: failed to start containers: kubelet
```

**Comment by alena1108** [comment]
Can potentially have the same root cause as https://github.com/rancher/rancher/issues/16476

**Comment by alena1108** [comment]
Should be fixed in rancher/rancher:v2.1.2-rc8

**Comment by jiaqiluo** [comment]
Rancher v2.1.2-rc8 
HA(3 nodes)  Airgap setup

Tried adding different custom clusters:
- 3 node( 1 etcd, 1 control plane, 1 worker )  Success!
- 6 node( 3 etcd, 1 control plane, 2 worker)  Success!
- 1 node( all roles)  Success!

Verified fixed.
<img width="1687" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/48231592-e9e67300-e36b-11e8-9a60-7d3732283445.png">
